sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -487,6 +488,43 @@ public Mono<Void> openConnectionsAndInitCaches() {
         }
     }
 
+
+    /***
+     *  Initializes the container by warming up the caches and connections for the current read region.
+     *
+     *  <p>
+     *  <br>NOTE: This API ideally should be called only once during application initialization before any workload.
+     *  <br>In case of any transient error, caller should consume the error and continue the regular workload.
+     *  </p>
+     *
+     * @return A String representative of open connections result.
+     */
+    @Beta(value = Beta.SinceVersion.V4_29_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public Mono<String> openConnectionsAndInitializeCaches() {
```

Comment:
The main drive to create a new API here is to return the final statistics here, like how many are established, how many are failed to open. Should I do this?

Reply:
We do not expose those, it's internal data and given this is a best effort kind of attempt, it will be better if we just return a Mono<Void>.

Reply:
The new API will work similarly to the older one from the end user's perspective (non backward compatible change). Wonder why not just keeping the older and make it execute the new stuff; it also avoids another API discussion topic ;-)

Reply:
Or even better, return a response object that we can "anchor" the Cosmos Diagnostic strings, in case we need them to understand what executed and how.

Reply:
Discussed offline, for now just log.info the aggregated result

Reply:
Updated:
Reuse the current API and only log the aggregated results